### PR TITLE
AWS OIDC: List Security Groups

### PR DESCRIPTION
**Pull request by marcoandredinis**
Context: https://github.com/gravitational/teleport/issues/29317

When creating an EC2 Instance Connect Endpoint, we can optionally provide a list of Security Groups to apply to it.
The default SG for the VPC is used if none SG is provided. Sometimes the default is not what the users want.

This PR adds an endpoint that the UI will use to list possible SG for the current VPC.
The user will then one or more SG and those are going to be applied to the EC2 Instance Connect Endpoint

Demo:
```
... /integrations/aws-oidc/teleportdev/securitygroups
->
200
{
   "securityGroups":[
      {
         "Name":"launch-wizard-3",
         "SecurityGroupID":"sg-01f0c6606f58fb035"
      },
      {
         "Name":"launch-wizard-9",
         "SecurityGroupID":"sg-044891c291faa4648"
      },
      {
         "Name":"default",
         "SecurityGroupID":"sg-23aa8a58"
      },
      // ...
   ]
}
```

### CODE REVIEW

**Review by Joerger — APPROVED** [review]

**Comment by marcoandredinis** [comment]
@probakowski Can you please take a look when you get a chance?

File: lib/integrations/awsoidc/list_security_groups.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+// SecurityGroup is the Teleport representation of an EC2 Instance Connect Endpoint
+type SecurityGroup struct {
+	// Name is the Security Group name.
+	// This is just a friendly name and should not be used for further API calls
+	Name string
+
+	// SecurityGroupID is the security group ID.
+	// This is the value that should be used when doing further API calls.
+	SecurityGroupID string
```
Comment (r0mant) [inline lib/integrations/awsoidc/list_security_groups.go:55]: Nit: I'd consider renaming this field to just `ID`. Like the name field above is just `Name`, not `SecurityGroupName`.
```suggestion
	ID string
```

File: lib/web/ui/integration.go [hunk 2]
```diff
@@ -204,6 +204,27 @@ type AWSOIDCListEC2Response struct {
 	NextToken string `json:"nextToken,omitempty"`
 }
 
+// AWSOIDCListSecurityGroupsRequest is a request to ListSecurityGroups using the AWS OIDC Integration.
+type AWSOIDCListSecurityGroupsRequest struct {
+	// Region is the AWS Region.
+	Region string `json:"region"`
+	// VPCID is the VPC to filter EC2 Instance Connect Endpoints.
```
Comment (r0mant) [inline lib/web/ui/integration.go:211]:
```suggestion
	// VPCID is the VPC to filter security groups by.
```

File: lib/web/ui/integration.go [hunk 3]
```diff
@@ -204,6 +204,27 @@ type AWSOIDCListEC2Response struct {
...
+	VPCID string `json:"vpcId"`
+	// NextToken is the token to be used to fetch the next page.
+	// If empty, the first page is fetched.
+	NextToken string `json:"nextToken"`
+}
+
+// AWSOIDCListSecurityGroupsResponse contains a list of VPC Security Groups and a next token if more pages are available.
+type AWSOIDCListSecurityGroupsResponse struct {
+	// SecurityGroups contains the page of SecurityGroups
+	SecurityGroupss []awsoidc.SecurityGroup `json:"securityGroups"`
```
Comment (r0mant) [inline lib/web/ui/integration.go:221]: Extra `s` at the end.
```suggestion
	SecurityGroups []awsoidc.SecurityGroup `json:"securityGroups"`
```

**Review by r0mant — APPROVED** [review]

**Pull request merged**